source/Octopus.Tentacle/Kubernetes/InMemoryTentacleScriptLog.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Octopus.Tentacle.Contracts;
+
+namespace Octopus.Tentacle.Kubernetes
+{
+    public class InMemoryTentacleScriptLog
+    {
+        readonly List<ProcessOutput> logs = new();
+        
+        public void Verbose(string message)
+        {
+            lock (logs)
+                logs.Add(new ProcessOutput(ProcessOutputSource.Debug, message));
+        }
+
+        public void Info(string message)
+        {
+            lock (logs)
+                logs.Add(new ProcessOutput(ProcessOutputSource.StdOut, message));
+        }
+
+        public void Error(string message)
+        {
+            lock (logs)
+                logs.Add(new ProcessOutput(ProcessOutputSource.StdErr, message));
+        }
+
+        public IReadOnlyCollection<ProcessOutput> PopLogs()
```

Comment:
Are we happy to pop the logs like this? We'll lose these logs if there's an exception later on

Reply:
I'm comfortable with this